packages/effect/src/RateLimiter.ts

```diff
@@ -0,0 +1,18 @@
+import type { Effect } from "./Effect.js"
+import type { Duration } from "./index.js"
+import * as internal from "./internal/rateLimiter.js"
+import type { Scope } from "./Scope.js"
+
+/**
+ * Limits the number of calls to a resource to a maximum amount in some interval using the token bucket algorithm.
+ *
+ * Note that only the moment of starting the effect is rate limited: the number of concurrent executions is not bounded.
+ *
+ * Calls are queued up in an unbounded queue until capacity becomes available.
+ */
+export interface RateLimiter {
+  <R, E, A>(task: Effect<R, E, A>): Effect<R, E, A>
+}
+
+export const make = (limit: number, window: Duration.DurationInput): Effect<Scope, never, RateLimiter> =>
+  internal.make(max, interval)
```

Comment:
@hsubra89 Thanks for making the change, but I don't think the changes in the argument names were propagated to the actual implementation.

Reply:
LOL, I just blindly committed using the Github UI and didn't re-check since I was mobile. Apologies. Will fix :(

Reply:
Fixed properly this time.